HPCC4J-594 Add support for skipping tests

jpmcmu opened the pull request:
- Updated version of surefire plugin to support excludes file
- Added logic to generate an excludes file to GH actions

Signed-off-by: James McMullan James.McMullan@lexisnexis.com

<!-- Thank you for submitting a pull request to the hpcc4j project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues (https://track.hpccsystems.com/).
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 hpcc4j-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,and
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and check-off all the items that apply (after pull request has been created)
-->

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [ ] I have created a corresponding JIRA ticket for this submission
- [ ] My code follows the code style of this project.
  - [ ] I have applied the Eclipse code-format template provided.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [ ] The change has been fully tested:
  - [ ] This change does not cause any existing JUnits to fail.
  - [ ] I have include JUnit coverage to test this change
  - [ ] I have performed system test and covered possible regressions and side effects.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] This change fixes the problem, not just the symptom

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

drealeed commented:
another issue is the ecl generation code that generates the files failing for some reason against the 160 and other clusters. Rodrigo had me comment it out locally in order to get my tests to run. Do you know if your changes will address that too?

rpastrana commented:
Hi @drealeed those aren't necessarily related to this change, can you write up the executeECLScript problem? Thanks

rpastrana commented on dfsclient/pom.xml:
```diff
@@ -34,6 +34,7 @@
                 <value>org.hpccsystems.ws.client.TestResultNotifier</value>
               </property>
             </properties>
+            <excludesFile>../excluded-tests.txt</excludesFile>
```
it seems ok to have a single excluded-tests file at the hpcc4j level, but I could also see an argument for dedicated excluded-tests files per sub project.
what are the main drawbacks/advantages of the current implementation?

rpastrana commented on pom.xml:
```diff
@@ -45,7 +45,7 @@
     <maven.deploy.version>2.8.2</maven.deploy.version>
```
can we have a sample entry in the above excluded-tests.txt file?
Perhaps create a new test simply for the purpose of always excluding it via this file?
Also, does the github action mechanism append excluded tests entries, or does it overwrite the file?

rpastrana reviewed (commented):
@jpmcmu a few questions.

rpastrana commented on wsclient/src/test/java/org/hpccsystems/ws/client/platform/RampsDevRegressionTest.java:
```diff
@@ -6,9 +6,11 @@
 import org.junit.Test;
 import org.junit.experimental.categories.Category;
 
+@Ignore
```
@drealeed I'd like to integrate these tests into the standard wsclient tests, is there anything RAMPS specific being tested? If so, could you provide ECL to recreate the structures being testsed here?

drealeed replied:
No, this was just a way to get a really comprehensive test of all possible record structures; I picked ramps dev because it had a lot of files for the regression test and I had access to it.  And at the time the ecl errors occurred reading files from ramps dev so I pointed the test at the previously failing files to make sure they worked now.

drealeed replied:
Unfortunately the files tested in testFileMetadata() no longer exist. That test can be removed. We have similar tests in tardis for different hpcc ecl structures now.

rpastrana replied:
Unless those ecl structures are proprietary, I'd like to absorb as many tests as possible. Does it make sense to donate those tests given the fact we can preemptively create the target files via ecl scripts?

rpastrana replied:
@drealeed do you recall how test::wsclient::noecl was created? it seems like an edge case we should continue to test

jpmcmu committed:
HPCC4J-594 Add support for skipping tests

- Updated version of surefire plugin to support excludes file
- Added logic to generate an excludes file to GH actions
- Separated test subproject test runs in GH actions
- Removed old tests
- Fixed surefire upgrade issues
- Added excluded-tests.txt file with examples

Signed-off-by: James McMullan James.McMullan@lexisnexis.com

jpmcmu committed:
Code review changes
